music21/musicxml/xmlToM21.py

```diff
@@ -2630,18 +2633,19 @@ def xmlForward(self, mxObj: ET.Element):
         if durationText := strippedText(mxDuration):
             change = opFrac(float(durationText) / self.divisions)
 
-            # Create hidden rest (in other words, a spacer)
-            # old Finale documents close incomplete final measures with <forward>
-            # this will be removed afterward by removeEndForwardRest()
-            r = note.Rest(quarterLength=change)
-            r.style.hideObjectOnPrint = True
-            self.addToStaffReference(mxObj, r)
-            self.insertInMeasureOrVoice(mxObj, r)
+            if self.parent.parent.wasWrittenByFinale:
```

Comment:
One of the few times I wish Python were Javascript -- need to check for grandparent existence...

```
if (self.parent
    and self.parent.parent
    and self.parent.parent.wasWrittenbyFinale
):
```

Vote YES on [PEP 505](https://peps.python.org/pep-0505/)